engine/networking overhaul (#17176)
docker/docs: network/packet-filtering-firewalls.md

For system integrators who wish to build the Docker runtime into other applications, explore the [`moby` project](https://mobyproject.org/).

## Setting the default bind address for containers

By default, the Docker daemon binds published container ports to the `0.0.0.0`
address. When you publish a container's ports as follows:

```console
docker run -p 8080:80 nginx
```

This publishes port 8080 to all network interfaces on the host, potentially
making them available to the outside world. Unless you've disabled IPv6 at the
kernel level, the port gets published on both IPv4 and IPv6.

You can change the default binding address for published container ports so that
they're only accessible to the Docker host by default. To do that, you can
configure the daemon to use the loopback address (`127.0.0.1`) instead.
To do so, configure the `"ip"` key in the `daemon.json` configuration file:

```json
{
  "ip": "127.0.0.1"
}
```

> This changes the default binding port to `127.0.0.1` for published container

Reviewer: ```suggestion
This changes the default binding address to `127.0.0.1` for published container
```

```

Restarting `dockerd` daemon inserts the interface into the `docker` zone.

## Docker and ufw

[Uncomplicated Firewall](https://launchpad.net/ufw){: target="_blank" rel="noopener"}
(ufw) is a frontend that ships with Debian and Ubuntu,
and it lets you manage firewall rules. Docker and ufw use iptables in ways
that make them incompatible with each other.

When you publish a container's ports using Docker, traffic to and from that
container gets diverted before it goes through the ufw firewall settings.
Docker routes container traffic in the `nat` table, which means that packets

> are diverted before it reaches the `filter` table that ufw uses. Packets are

Reviewer: I think people are more used to talk about the INPUT/OUTPUT chains than the filter table. Moreover, the current wording is not 100% accurate, because NATed packets go through the FORWARD chain located in the filter table.
 
```suggestion
are diverted before it reaches the `INPUT` and `OUTPUT` chains used by ufw. Packets are
```

routed before the firewall rules can be applied, effectively ignoring your

> firewall configuration.

Reviewer 2: I think this advice also applies to firewalld? (_differs a little bit with it's zones_)

A UFW specific issue that differs behaviour with Docker is this: https://github.com/moby/moby/issues/27817
